nixos/modules/services/networking/uptermd.nix

```diff
@@ -0,0 +1,81 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.uptermd;
+in
+{
+  options = {
+    services.uptermd = {
+      enable = mkEnableOption "uptermd";
+
+      openPorts = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to open firewall ports for upterm";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 2222;
+        description = ''
+          Port the server will listen on.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        example = "[::]";
```

Comment:
Can't we listen on ipv4 and ipv6?

Reply:
In both cases (listenAddress = "0.0.0.0", listenAddress = "[::]") the uptermd process opens its sockets to both protocols.
```
# ss -tlpn | grep upterm
LISTEN 0      4096                       *:2222             *:*    users:(("uptermd",pid=996,fd=3))
```

In case that the process only listens to one protocol, "[::]" (IPv6) would be more preferable, because the Linux kernel is able to represent incoming IPv4 connections as IPv6 connections. This is called [IPv4-mapped IPv6 addresses](https://tools.ietf.org/html/rfc3493#section-3.7).

Reply:
Using `net.ipv6.bindv6only` and defaulting to `[::]` is the sane way to default to a dualstacked bind.

Reply:
I have changed the default listenAddress to `[::]` because I don't see any usecase for binding to localhost.